native/packages/react-native-bpk-component-button/src/BpkButton-test.common.js

```diff
@@ -125,6 +125,32 @@ const commonTests = () => {
         iconOnly: true,
       }, 'icon', 'BpkButton').toString()).toEqual('Error: Invalid prop `icon` supplied to `BpkButton`. When `iconOnly` is enabled, `icon` must be supplied.'); // eslint-disable-line max-len
     });
+
+    it('should reject theme property when button type is not themeable', () => {
+      expect(propTypes.theme({
+        type: 'featured',
+        theme: {},
+      }, 'theme', 'BpkButton').toString()).toEqual('Error: Invalid prop `theme` supplied to `BpkButton`. `type` is `featured`, but themeable types are `primary, secondary`.'); // eslint-disable-line max-len
```

Comment:
Hmm, i think this might be a little overzealous. Imagine a scenario where an app has multiple button types inside. A `<ThemeProvider theme={myTheme} />` is wrapped around the entire app. This would throw a warning when in fact the use case is valid.

Reply:
That's a good point yeah. Shall we just have it 'fail' silently?

Reply:
I've removed warning when the type isn't themeable. Currently the warning when they fail to provide all the necessary theme attributes is still there - can remove it if we think that's overzealous too. I'm just concerned that failing silently (by falling back to no theme) will be a bad UX for consumers. What do you think?

Reply:
So we defo had to remove the other one because it would throw a warn when the consumer use case was completely valid. Our incomplete theme object warning shouldn't suffer the same problem as an incomplete theme is...well just that - an incomplete theme. I cannot think of a case where supplying an incomplete theme is beneficial, therefore I think our warning is good.